internal/pkg/util/exec/starter_linux.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package exec
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+// LookStarterPath returns the starter binary path and also returns
+// if the path point to the setuid starter executable
+func LookStarterPath(suid bool, allowedSuid bool) (string, bool) {
```

Comment:
this function should not be exported.

Since you have two booleans, strictly speaking you have four functions. That does not seem to be the case, though, as allowedSuid is only used if suid is true, so that's three functions (`(true, true)`, `(true, false)` and `(false, *)`).

The last one is easy: `GetStarterPath`, this calls `getStaterPath(false, false)`.

The other two are `GetStarterSuidPath`, this calls `getStaterPath(true, *)`.

Also, the second return value should be an error: if you ask for starter-suid and you don't get it, you should get an error, not the path to starter.